app/domain/finders/content.rb

```diff
@@ -22,42 +22,75 @@ def call
   ALL = 'all'.freeze
   NONE = 'none'.freeze
 
-  attr_reader :organisation_id, :document_type, :date_range, :search_term
+  attr_reader :filter
 
   def initialize(filter)
-    @organisation_id = filter.fetch(:organisation_id)
-    @document_type = filter.fetch(:document_type)
-    @search_term = parse_search_term(filter[:search_term]) if filter[:search_term]
-    @page = filter[:page] || 1
-    @page_size = filter[:page_size] || DEFAULT_PAGE_SIZE
+    @filter = filter
+
+    @page = filter.fetch(:page, 1)
+    @page_size = filter.fetch(:page_size, DEFAULT_PAGE_SIZE)
     @date_range = filter.fetch(:date_range)
     @sort_attribute = filter.fetch(:sort_attribute) || 'upviews'
     @sort_direction = filter.fetch(:sort_direction) || 'desc'
   end
 
   def view
-    @view ||= Finders::SelectView.new(date_range).run
+    @view ||= Finders::SelectView.new(filter[:date_range]).run
+  end
+
+  def do_filter
+    scope = view[:model_name].all
+    scope = find_by_search_term(scope)
+    scope = find_by_document_type(scope)
+    scope = find_by_organisation(scope)
```

Comment:
Could we add the conditionals about applying filters within this method, rather than the individual methods? I feel it will make the logic easier to read, as it currently reads that filters are **always** being applied.

Reply:
I have tried, but it does not read as well (although this is totally subjective). If you have a strong view I will do it, but can you please confirm that you would rather the following?:

```ruby
  def apply_filter
    scope = view[:model_name].all
    scope = find_by_search_term(scope) if filter[:search_term].present?
    scope = find_by_document_type(scope) if filter[:document_type].present?
    scope = find_by_organisation(scope)

    scope
  end
```

Reply:
I'd rather this, as it will also improve the readability of the individual functions themselves.

Reply:
I have updated it. I honestly can't see the obvious benefit, but I would rather updating it and not having a long discussion about it.